skyportal/handlers/api/shift.py

```diff
@@ -448,63 +458,66 @@ def patch(self, shift_id, user_id):
         except ValueError:
             return self.error("Invalid userID parameter")
 
-        shiftuser = (
-            ShiftUser.query_records_accessible_by(self.current_user, mode='update')
-            .filter(ShiftUser.shift_id == shift_id)
-            .filter(ShiftUser.user_id == user_id)
-            .first()
-        )
-
-        if shiftuser is None:
-            return self.error(f"User {user_id} is not a member of shift {shift_id}.")
+        with self.Session() as session:
+            shiftuser = session.scalars(
+                ShiftUser.select(session.user_or_token, mode='update')
+                .where(ShiftUser.shift_id == shift_id)
+                .where(ShiftUser.user_id == user_id)
+            ).first()
 
-        admin = data.get("admin", shiftuser.admin)
-        if not isinstance(admin, bool):
-            return self.error(
-                "Invalid (non-boolean) value provided for parameter `admin`"
-            )
-        shiftuser.admin = admin
+            if shiftuser is None:
+                return self.error(
+                    f"User {user_id} is not a member of shift {shift_id}."
+                )
 
-        needs_replacement = data.get("needs_replacement", False)
-        try:
-            needs_replacement = bool(needs_replacement)
-        except (ValueError, TypeError):
-            return self.error(
-                "Invalid needs_replacement parameter: unable to parse to boolean"
-            )
-        shiftuser.needs_replacement = needs_replacement
+            admin = data.get("admin", shiftuser.admin)
+            if not isinstance(admin, bool):
+                return self.error(
+                    "Invalid (non-boolean) value provided for parameter `admin`"
+                )
+            shiftuser.admin = admin
 
-        if needs_replacement:
-            # send a user notification to all members of the group associated to the shift
-            # that the user needs to be replaced
-            # recover all group users associated to the shift
+            needs_replacement = data.get("needs_replacement", False)
             try:
-                shift = Shift.get_if_accessible_by(
-                    shift_id,
-                    self.current_user,
-                    options=[
-                        joinedload(Shift.group).joinedload(Group.group_users),
-                    ],
-                    raise_if_none=True,
+                needs_replacement = bool(needs_replacement)
+            except (ValueError, TypeError):
+                return self.error(
+                    "Invalid needs_replacement parameter: unable to parse to boolean"
```

Comment:
This should also be a test if boolean, not a cast into boolean.

Reply:
good idea.